src/schema/ecommerce/types/order_line_item.ts

```diff
@@ -21,11 +21,11 @@ export const OrderLineItemType = new GraphQLObjectType({
       type: Artwork.type,
       description: "Artwork that is being ordered",
       resolve: (
-        { artworkId },
+        { artworkVersionId },
         _args,
         _context,
-        { rootValue: { authenticatedArtworkLoader } }
-      ) => authenticatedArtworkLoader(artworkId),
+        { rootValue: { authenticatedArtworkVersionLoader } }
```

Comment:
this endpoint returns an `ArtworkVersion` (https://github.com/artsy/gravity/blob/master/app/api/v1/artwork_versions_endpoint.rb#L13) but its being resolved to an `Artwork` in https://github.com/artsy/metaphysics/pull/1326/files#diff-03ea9cde229e19c3db2c2640d865c49aL21, I think we can do two things,
- either have artwork version return actual artwork and resolve this with `artwork_version.artwork`
or
- add a new `ArtworkVersion` type which matches to pretty much https://github.com/artsy/gravity/blob/master/app/models/domain/artwork_version.rb#L2 and just return that

I think having `ArtworkVersion` as a new type makes more sense to me.

Reply:
Yeah, I wonder if this should return something like:

```graphql
type ArtworkVersion {
  artwork: Artwork
  version: ArtworkVersionMetadata
}

type Query {
  authenticatedArtworkVersionLoader(ID: ID): ArtworkVersion
}
```

And MP composes both the artwork and the version metadata  in 
https://github.com/artsy/gravity/blob/master/app/models/domain/artwork_version.rb#L2

So that you can just ignore the version metadata when you don't need it (which at a guess, is most of the time) ?

Reply:
That may break existing queries though, so unsure if that's a blocker.

Another option is to make a separate resolver that specifically claims to only return the versioned artwork, instead of an ArtworkVersion from grav?

Reply:
Because we're not sure how versions will get used across the platform in the future, I'd prefer a design that doesn't mingle the version object's data into artworks and instead treats it as a separate, first-class thing.

Most uses can continue to reference "artworks" (which implicitly refers to currently-published artwork data), while any uses that want versions would make that explicit, with similar (but not necessarily identical) fields available as on artworks.

Reply:
Yeah, after chatting with Ashkan - I agree, we should make a new type in MP for this

Reply:
Yeah, it should be it's own type. We can also add the current version, and a list of all versions as fields on the artwork type.